### PR TITLE
STR-57--BE:Send an email to the "Client" when a student becomes assigned to a group

### PR DESCRIPTION
**Pull request by mbendandour**
# Description
Sends an email as soon as a new user is assigned a role  
# Testing
Since the .env file 'MAIL_DRIVER' in my environment is set to log, I check my storage/logs to see if the email was actually sent. If your mail Driver is set to 'smtp' you should check if an email was received in whatever email you designated to RECEIVE_EMAIL

-Log into the app. Since 'steve@csun.edu' is an admin, the user group will be null so don't sign in with steve. Make sure whatever email you use to log in is assigned a research id in the **research_id** table. Log in with username 'matt' and no password
-Once you log in as a new user, check your email (smtp) or storage/log (log) to see if email was received. Email should contain user group and research id
# Installation
'Are there any dependencies the reviewer needs to download?'

### CODE REVIEW

File: app/Services/UserAssignedGroupEmailService.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+/**
```
Comment (frlzjosh) [inline app/Services/UserAssignedGroupEmailService.php]: are you able to take these comments out ?

File: app/Services/UserGroupService.php [hunk 2]
```diff
@@ -5,12 +5,20 @@
 namespace App\Services;
 
 use App\Contracts\UserGroupContract;
+use App\Http\Controllers\MailController;
```
Comment (frlzjosh) [inline app/Services/UserGroupService.php]: I don't think that we need to use the controller here? doesn't that go against the contract/services flow ?

File: tests/phpunit/ServicesTests/UserGroupServiceTest.php [hunk 3]
```diff
@@ -4,31 +4,38 @@
...
-
+use App\Contracts\UserGroupContract;
+Use Mockery;
 class UserGroupServiceTest extends TestCase
 {
     use DatabaseMigrations;
 
     protected $utility;
+    protected $userGroupUtility;
+    //protected $userAssignedGroupEmailUtility;
```
Comment (frlzjosh) [inline tests/phpunit/ServicesTests/UserGroupServiceTest.php]: remove the comment

File: tests/phpunit/ServicesTests/UserGroupServiceTest.php [hunk 4]
```diff
@@ -55,7 +62,7 @@ public function getGroup_returns_group_remember_token_display_name_and_email_fro
      */
     public function user_is_evenly_distributed_into_the_groups_and_assigned_a_group()
     {
-        $service = new UserGroupService();
+       // $service = new UserGroupService($this->userAssignedGroupEmailUtility);
```
Comment (frlzjosh) [inline tests/phpunit/ServicesTests/UserGroupServiceTest.php]: can u remove the commented code

**Review by frlzjosh — DISMISSED** [review]
very small changes. the functionality works

File: app/Mail/GenericEmail.php [hunk 5]
```diff
@@ -29,5 +33,7 @@ public function __construct()
     public function build()
     {
         return $this->view('mail.generic');
+//        return $this->from('donotreply@example.com')
```
Comment (frlzjosh) [inline app/Mail/GenericEmail.php]: can we also get rid of this line of code ?

File: tests/phpunit/ServicesTests/UserGroupServiceTest.php (hunk 3 above)
Comment (liadGolan) [reply tests/phpunit/ServicesTests/UserGroupServiceTest.php]: this line should be uncommented but still present

File: routes/web.php [hunk 6]
```diff
@@ -20,7 +20,6 @@
 
 Route::get('/get_group/{user_id}','UserGroupController@getGroup');
 
-Route::get('/sendMailTest','MailController@sendMail');
```
Comment (liadGolan) [inline routes/web.php]: This should not get deleted

**Review by liadGolan — CHANGES_REQUESTED** [review]

**Review by liadGolan — APPROVED** [review]
👍

File: app/Providers/UserAssignedGroupEmailServiceProvider.php [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<?php
+/**
```
Comment (frlzjosh) [inline app/Providers/UserAssignedGroupEmailServiceProvider.php]: take out comment

File: app/Http/Controllers/MailController.php [hunk 8]
```diff
@@ -2,16 +2,27 @@
...
+        $this->emailUtility = $emailUtility;
+    }
 
-        return "good";
+    public function sendMail()
+    {
+        return $this->emailUtility->sendMail();
     }
 }
+//create a service and contract and call the contract in the user group service. then test that
```
Comment (frlzjosh) [inline app/Http/Controllers/MailController.php]: take out comment

File: app/Contracts/UserAssignedGroupEmailContract.php [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<?php
+/**
```
Comment (frlzjosh) [inline app/Contracts/UserAssignedGroupEmailContract.php]: take out comment

**Review by frlzjosh — CHANGES_REQUESTED** [review]
just a few more comments that you need to get rid of

**Review by nercesk — APPROVED** [review]

**Review by frlzjosh — APPROVED** [review]
It is good!

**Pull request merged**